### ISSUE TITLE
GeoIP folders missing on new install

### ISSUE THREAD

**Issue by GiorgioAresu** [opnsense/core#4704]
**Important notices**

Before you add a new report, we ask you kindly to acknowledge the following:

- [x] I have read the contributing guide lines at https://github.com/opnsense/core/blob/master/CONTRIBUTING.md
- [x] I have searched the existing issues and I am convinced that mine is new.

**Describe the bug**

First time using OPNsense, on a new 21.1 install upgraded to 21.1.1 immediately after, the entire `/usr/local/share/GeoIP` folder was missing, leading to failures configuring GeoIP which took me a while to identify.

A `mkdir -p /usr/local/share/GeoIP/alias` solved all.

I wasn't able to find anything about this on the documentation nor internet, just found old threads on the forum which helped understanding more.

**Environment**

OPNsense 21.1.1-amd64 on apu2e4 (installed from OPNsense-21.1-OpenSSL-serial-amd64)
Verified on OPNsense-21.1-OpenSSL-vga-amd64 live on an intel NUC

**Comment by AdSchellevis** [comment]
The missing directory is unlikely the issue, if doesn't exist it will create it on download.
https://github.com/opnsense/core/blob/de291e62660996b04d589381cad3396db5e9eb80/src/opnsense/scripts/filter/lib/geoip.py#L105-L106

(tried `rm -rf /usr/local/share/GeoIP` and apply aliases on my which seems to work fine)

**Comment by GiorgioAresu** [comment]
Ok, so I tried removing the folder on my install and it recreated them just fine. I tried the live image again, and configured the correct url, hit apply and it created them correctly. May be a combination of wrong url (.tar.gz suffix) and/or wan not connected when I first tried to configure them on my current install?
As I said it was a fresh install, no config imported, and it somehow happened, so I think there is a bug. At this point I'm not sure how, though.
If I can I'll try again with the live of the serial image on the apu2 at a time I can bring down the network

**Comment by GiorgioAresu** [comment]
To add informations, yesterday I was trying forcing the refresh with
```
# cd /usr/local/opnsense/scripts/filter/lib
# python3
>>> from geoip import download_geolite
>>>  download_geolite()
```
And I got this error (found the search in my browser history)
**No such file or directory: '/usr/local/share/GeoIP/alias.stats'**

Then I created the /usr/local/share/GeoIP folder, and the error changed to the JSON saying address count: 0, can't find the exact response. At this point using the correct url didn't work, the only thing that fixed it was creating the alias subfolder

**Comment by AdSchellevis** [comment]
it needs settings first, apply via the gui should flush the config template and refresh the aliases

**Comment by GiorgioAresu** [comment]
Yep, that's what I did. The main reason I noticed something was wrong with it was because all alias were broken and so rules with them were not working. Same rule with ip was working, rule with that ip in an alias (any of them) wasn't. pfTables was showing only bogus alias had any entries

**Comment by NightFlexer** [comment]
on OPNsense 25.7 the problem is also present
